### PR TITLE
CNV consensus (2 of n): Filter raw data of CNV call methods

### PR DESCRIPTION
**Pull request by fingerfen**
### Purpose/implementation Section

To add to the copy number consensus call's analysis

#### What GitHub issue does your pull request address?

Issue #128 

#### Which areas should receive a particularly close look?

Whether the code works, comment is good, and if awk command is not too convoluted

#### Is there anything that you want to discuss further?

No

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

### CODE REVIEW

**Comment by cgreene** [comment]
Hi @fingerfen : do you recognize this error?
```
KeyError in line 5 of /rocker-build/analyses/copy_number_consensus_call/Snakefile:
'samples'
  File "/rocker-build/analyses/copy_number_consensus_call/Snakefile", line 5, in <module>
```

I added some lines to list the contents of the scratch directory on CI and then also print out the sample keys to make sure they were in the file. Those appear to be present:
https://circleci.com/gh/AlexsLemonade/OpenPBTA-analysis/1606?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: analyses/copy_number_consensus_call/run_consensus_call.sh [hunk 1]
```diff
@@ -18,3 +18,25 @@ python3 src/scripts/merged_to_individual_files.py \
...
+## Run the Snakemake pipeline
+## -s : Point to the location of the Snakemake file
+## --configfile : Point to the location of the config file
+## -d : Specifying working directory. The "." after -d is to show the working dir is at the current folder
+## -j : Set available cores, in this case, when no number is provided, thus use all available cores
+## -p : Print shell command that will be executed
+## --restart-times : Define the times a job restarts when run into an error before giving up
+## --latency-wait: Define the number of seconds to wait for a file to show up after that file has been created
+
+snakemake \
```
Comment (jashapiro) [inline analyses/copy_number_consensus_call/run_consensus_call.sh]:
```suggestion
snakemake \
```

Comment (jashapiro) [reply analyses/copy_number_consensus_call/run_consensus_call.sh]: Fixed by the commit below. https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/313/commits/d000a1b710745673b9be25ff430335d63059d037

Comment (fingerfen) [reply analyses/copy_number_consensus_call/run_consensus_call.sh]: Thank you very much for the help! It was my mistake. I inserted a space accidentally.

Comment (jashapiro) [reply analyses/copy_number_consensus_call/run_consensus_call.sh]: No problem! That is why we do the CI and code review!

File: analyses/copy_number_consensus_call/Snakefile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    output:
+        ## Define the output files' names
+        freec_del="../../scratch/interim/{sample}.freec.del.bed",
+        freec_dup="../../scratch/interim/{sample}.freec.dup.bed"
+    threads: 1
+    params:
+        ## Take parameters from the config file and assign them into params for convinient use in the Shell section
+        SIZE_CUTOFF=str(config["size_cutoff"]),
+        FREEC_PVAL=str(config["freec_pval"]),
+        mem=r'1G'
```
Comment (cgreene) [inline analyses/copy_number_consensus_call/Snakefile]: What does this line `mem=...` do? The snakemake docs that I found have this in a resources section and use `mem_mb`

Comment (fingerfen) [reply analyses/copy_number_consensus_call/Snakefile]: This line specifies the **memory usage** for this specific task **on the cluster**. So for this task, Snakemake is to use 1G of memory to complete it on the cluster. One can make use of the `mem=...` and `threads` by adding
`-c "qsub -l h_vmem={params.mem} -l mem_free={params.mem} -pe smp {threads} -V -cwd"`
to the Snakemake command in the `run_consensus_call.sh` file

However, for this pull request, I designed this to work on the local system so I didn't include the `-c` flag in the `run_consensus_call.sh` file. Thus, the `men=...` is not being used

File: analyses/copy_number_consensus_call/run_consensus_call.sh [hunk 3]
```diff
@@ -18,3 +18,25 @@ python3 src/scripts/merged_to_individual_files.py \
     --cnvkit ../../data/pbta-cnv-cnvkit.seg.gz \
     --freec ../../data/pbta-cnv-controlfreec.tsv.gz \
     --snake ../../scratch/config_snakemake.yaml
+
+ls ../../scratch/
```
Comment (jashapiro) [inline analyses/copy_number_consensus_call/run_consensus_call.sh]: We can probably delete this line now that we found the bug...

File: analyses/copy_number_consensus_call/run_consensus_call.sh [hunk 4]
```diff
@@ -18,3 +18,25 @@ python3 src/scripts/merged_to_individual_files.py \
...
+cat ../../scratch/config_snakemake.yaml
+
+## Run the Snakemake pipeline
```
Comment (jashapiro) [inline analyses/copy_number_consensus_call/run_consensus_call.sh]: Maybe here describe here what the snakemake pipeline is doing, in broad strokes.

**Comment by fingerfen** [comment]
Thank you for the help @cgreene and @jashapiro. I have made all the changes and removed the `mem=` and `threads` from the Snakefile

**Pull request merged**